src/lib/metrics.js

```diff
@@ -0,0 +1,334 @@
+  },
+
+  installStart() {
+    if (installTimer) {
+      clearTimeout(installTimer);
+    }
+    installTimer = setTimeout(() => {
+      this.installFailure();
+    }, INSTALL_FAILURE_DELAY);
+
+    this.sendEvent({
+      ec: 'install-addon',
+      ea: 'button-click',
+      el: 'install-trigger',
+      cm1,
+      cm2,
+      cm3,
+      cd3,
+      cd4,
+      cd5,
+      cd6,
+      cd7,
+      cd8,
+      cd9,
+      cd10,
+      cd11
+    });
+  },
+
+  installFailure() {
+    this.sendEvent({
+      ec: 'install-addon',
+      ea: 'poll-event',
+      el: 'install-fail'
+    });
+  },
+
+  installSuccess() {
+    if (!installTimer) {
+      // Skip this event if there's no install timer in progress.
+      return;
+    }
+    clearTimeout(installTimer);
+    this.sendEvent({
+      ec: 'install-addon',
+      ea: 'poll-event',
+      el: 'install-success'
+    });
+  },
+
+  themeChangeFull(themeId) {
+    cm1++;
+    this.setThemeChanged(true);
+    this.sendEvent({
+      ec: 'theme-change',
+      ea: 'select-full',
+      el: themeId,
+      cm1,
+      cm2,
+      cm3,
+      cd1,
+      cd2
+    });
+  },
+
+  themeChangeBackground(backgroundId) {
+    cm2++;
+    this.setThemeChanged(true);
+    this.sendEvent({
+      ec: 'theme-change',
+      ea: 'select-background',
+      el: backgroundId,
+      cm1,
+      cm2,
+      cm3,
+      cd1,
+      cd2,
+      cd3,
+      cd11
+    });
+  },
+
+  themeChangeColor(colorName) {
+    cm3++;
+    this.setThemeChanged(true);
+    this.sendEvent({
+      ec: 'theme-change',
+      ea: 'select-color',
+      el: colorName,
+      cm1,
+      cm2,
+      cm3,
+      cd1,
+      cd2,
+      cd3
+    });
+  },
+
+  shareClick() {
+    this.sendEvent({
+      ec: 'share-engagement',
+      ea: 'button-click',
+      cm1,
+      cm2,
+      cm3,
+      cd1,
+      cd2,
+      cd3,
+      cd4,
+      cd5,
+      cd6,
+      cd7,
+      cd8,
+      cd9,
+      cd10,
+      cd11
+    });
+  },
+
+  linkClick(el) {
+    // if el === download-firefox, add the following dimensions to this event
+    const downloadFirefoxDimensions =
+      el !== 'download-firefox' ? {} : { cd5, cd6, cd7, cd8, cd9, cd10, cd11 };
+    this.sendEvent({
+      ec: 'link-engagement',
+      ea: 'click',
+      el,
+      cm1,
+      cm2,
+      cm3,
+      cd1,
+      cd2,
+      cd3,
+      cd4,
+      ...downloadFirefoxDimensions
+    });
+  },
+
+  receiveTheme(action, theme) {
+    this.sendEvent({
+      ec: 'receive-theme',
+      ea: 'button-click',
+      el: action,
+      cd1,
+      ...this.themeToDimensions(theme)
+    });
+  },
+
+  finishVisit() {
+    this.sendEvent({
+      ec: 'finish-visit',
+      ea: 'leave',
+      cm2,
+      cm3,
+      cd1,
+      cd2,
+      cd3,
+      cd4,
+      cd5,
+      cd6,
+      cd7,
+      cd8,
+      cd9,
+      cd10,
+      cd11
+    });
+  },
+
+  sendEvent(params) {
+    if (!GA_TRACKING_ID || !clientUUID || navigator.doNotTrack === 1) {
```

Comment:
We check `navigator.doNotTrack` here, but below on L317, we do an explicit check if `typeof navigator !== 'undefined'`. Do we need to check if `navigator` and `navigator.doNotTrack` are kosher here before checking the value?

Reply:
I think that's just a check left over from some code I copied out of what had been used in a legacy add-on (and thus no `navigator` global). I can probably just remove the check on L317